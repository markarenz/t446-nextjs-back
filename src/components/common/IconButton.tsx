import styles from '../../styles/modules/IconButton.module.scss';

type Props = {
  children: JSX.Element;
  onClick: Function;
  title: string;
  color?: string;
  disabled?: boolean;
};
const IconButton: React.FC<Props> = ({
  children,
  onClick,
  title,
  color,
  disabled
}) => {
  let classColor = styles.default;
  if (color === 'secondary') {
    classColor = styles.secondary;
  }
  if (color === 'primary') {
    classColor = styles.primary;
  }
  const optionalProps: any = {};
  if (disabled !== undefined) {
    optionalProps.disabled = disabled;
  }
  return (
    <button
      type="button"
      title={title}
      onClick={() => onClick()}
      className={`${styles.iconButton} ${classColor}`}
      {...optionalProps}
    >
      <div className={styles.inner}>{children}</div>
    </button>
  );
};

export default IconButton;
