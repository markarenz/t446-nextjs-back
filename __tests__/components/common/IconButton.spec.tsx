import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import IconButton from '../../../src/components/common/IconButton';

const defaultProps = {
  onClick: jest.fn(),
  title: 'Test',
  color: 'primary',
  disabled: false
};
const altProps = {
  ...defaultProps,
  color: 'secondary'
};
describe('Button', () => {
  it('renders default button', async () => {
    render(
      <IconButton {...defaultProps}>
        <span>Test</span>
      </IconButton>
    );
    const element = await screen.findByRole('button');
    expect(element).toBeInTheDocument();
  });
  it('renders button with secondary color and outlines', async () => {
    render(
      <IconButton {...altProps}>
        <span>Test</span>
      </IconButton>
    );
    const element = await screen.findByRole('button');
    expect(element).toBeInTheDocument();
  });

  it('fires onClick when clicked', async () => {
    render(
      <IconButton {...defaultProps}>
        <span>Test</span>
      </IconButton>
    );
    const element = await screen.findByRole('button');
    fireEvent(
      element,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true
      })
    );
    expect(defaultProps.onClick).toHaveBeenCalled();
  });
});
