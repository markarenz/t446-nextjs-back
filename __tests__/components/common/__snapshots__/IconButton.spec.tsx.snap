// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders default button 1`] = `
<button
  class="iconButton primary"
  title="Test"
  type="button"
>
  <div
    class="inner"
  >
    <span>
      Test
    </span>
  </div>
</button>
`;
